### PR TITLE
Bug: fix session timeout handling post submission

### PR DESCRIPTION
**Pull request by neil-stephen-mcgonigle**
- refactored keep-alive tests out to own file
- added post submission sessionResource handling

### CODE REVIEW

File: questionnaire/questionnaire-service.js [hunk 1]
```diff
@@ -350,7 +350,23 @@ function createQuestionnaireService({
         return progressEntryResource;
     }
 
+    function applicationSubmitted(submissionStatus) {
+        return submissionStatus !== 'NOT_STARTED';
```
Comment (armoj) [inline questionnaire/questionnaire-service.js:354]: does this _categorically_ mean that it is submitted?

File: questionnaire/questionnaire-service.js [hunk 2]
```diff
@@ -350,7 +350,23 @@ function createQuestionnaireService({
...
     async function buildSessionInformationBlock(questionnaireId) {
+        const submissionStatus = await getQuestionnaireSubmissionStatus(questionnaireId);
+
+        if (applicationSubmitted(submissionStatus)) {
+            const dateNow = Date.now();
+            return {
+                alive: false,
+                duration: 0,
+                created: dateNow,
+                expires: dateNow
```
Comment (armoj) [inline questionnaire/questionnaire-service.js:366]: not understanding the reason behind these 3 properties. duration should not be 0, and it was created before now.

**Review by armoj — CHANGES_REQUESTED** [review]

File: questionnaire/questionnaire-service.js (hunk 1 above)
Comment (neil-stephen-mcgonigle) [reply questionnaire/questionnaire-service.js:354]: As far as I am aware the application has been submitted in the sense that the Agree and submit" button has been clicked the application now no longer has a submission_status of 'NOT_STARTED' so submission HAS started the possible states post submission application could be in are [FAILED, IN_PROGRESS, COMPLETE].  
All of which represent post submission status therefore a confirmation page will be shown and the application session will be over.

File: questionnaire/keep-alive.test.js [hunk 3]
```diff
@@ -0,0 +1,340 @@
+'use strict';
+
+/* * ****************************************************************************************** * */
+/* * ****************************************************************************************** * */
+/* *         THIS FILE IS GENERATED. ALL MANUAL EDITS MADE TO THIS FILE WILL BE LOST!!!         * */
+/* *         --------------------------------------------------------------------------         * */
+/* *         If you need to make a change to this test file you will need to edit the           * */
+/* *         generate-tests file and regenerate the tests using command line.                   * */
+/* * ****************************************************************************************** * */
+/* * ****************************************************************************************** * */
```
[the remark this answers is not in the document]
Comment (neil-stephen-mcgonigle) [reply questionnaire/keep-alive.test.js:10]: A hangover from the copy and paste, I shall also remove from here 
https://github.com/CriminalInjuriesCompensationAuthority/data-capture-service/blob/c818f6f9e40fe132da42dcfe50a2f61930b4df95/questionnaire/questionnaire.test.js#L4

File: questionnaire/questionnaire-service.js (hunk 2 above)
Comment (neil-stephen-mcgonigle) [reply questionnaire/questionnaire-service.js:366]: These properties represent the state of _this_ session resource, the duration is 0 seconds as the session has ended, the resource is being created and expired at the same time. In essence all parameters represent an invalid session expiry resource.

Comment (neil-stephen-mcgonigle) [reply questionnaire/questionnaire-service.js:366]: happy to discuss also happy to change this from dateNow to creationDate or equivalent

File: questionnaire/questionnaire-service.js (hunk 1 above)
Comment (neil-stephen-mcgonigle) [reply questionnaire/questionnaire-service.js]: Made the change anyway, removed the function and added the conditional submissionStatus check within the code.

**Review by armoj — APPROVED** [review]

**Pull request merged**